### PR TITLE
WIP: glean: Avoid returning duplicate sequence numbers

### PR DESCRIPTION
**Pull request by mdboom**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: components/service/glean/src/main/java/mozilla/components/service/glean/ping/PingMaker.kt [hunk 1]
```diff
@@ -39,12 +39,17 @@ internal class PingMaker(
      * @param pingName The name of the ping
      * @return sequence number
      */
+    @Synchronized
```
Comment (Dexterp37) [inline components/service/glean/src/main/java/mozilla/components/service/glean/ping/PingMaker.kt]: This makes sense to me, in theory. However, I'm struggling to figure out a case in which that would happen :\

Comment (mdboom) [reply components/service/glean/src/main/java/mozilla/components/service/glean/ping/PingMaker.kt]: Yeah, ditto.  I'm grasping at straws, honestly.  Also, it introduces an interkitten in Taskcluster that I can't reproduce locally...

Comment (Dexterp37) [reply components/service/glean/src/main/java/mozilla/components/service/glean/ping/PingMaker.kt]: @travis79 @mdboom .. any idea?

Comment (travis79) [reply components/service/glean/src/main/java/mozilla/components/service/glean/ping/PingMaker.kt]: Maybe if we we hit the events limit, went over by 1 or more events, and then immediately went to background.  All other paths through here seem pretty isolated/sync.   That's the only situation I can come up with that might have us two separate execution threads in this same bit of code near the same time.

Comment (mdboom) [reply components/service/glean/src/main/java/mozilla/components/service/glean/ping/PingMaker.kt]: Yeah.  I can't imagine anything else.  But that seems really unlikely to explain the high number of the dupes found in chutten's analysis.  (Plus, I'm not sure any of the dupes involve the event ping).  I also don't think we're getting a ton of events from fenix, but it's probably worth checking to see if we have any event pings with around 500 events in them.

**Pull request closed without merging**